test/test_torch.py

```diff
@@ -17838,407 +17839,435 @@ def inner(self, device, dtype):
 # - (sub)test name (string)
 # - tensor constructor, takes dtype and device and constructs the tensor to run the op on
 # - arg constructor, takes dtype and device and constructs op arguments
-# - torch.half precision (=1e-5)
-# - torch.bfloat16 precision (=1e-5)
-# - precision (=1e-5), precision to use for all other dtypes
 # - dtype_list (=_types), a list of torch dtypes to test the op(s) with
 # - cpu_dtype_list (=[]), a list of torch dtypes to test the op(s) on cpu
 # - make_inplace_variant (=True), if true the inplace version of the op (op_) is also tested
 # - decorators (=[]), a list of decorators to apply to the test
+# - half_tol (=None, None) override (rtol,atol) for half tests
+# - win_float_tol (=None, None) override (rtol,atol) for float tests on windows
+class _TestOperatorMeta():
+
+    inplace = None
+
+    def __init__(self, op_name, subtest_name='', ctor=None,
+                 arg_ctor=None, dtypes=None, cpu_dtypes=None,
+                 has_inplace=True, decorators=None, *, half_tol=(None, None), win_float_tol=(None, None)):
+
+        if has_inplace:
+            self.inplace = _TestOperatorMeta(op_name + '_', 'inplace_' + subtest_name,
+                                             ctor=ctor, arg_ctor=arg_ctor,
+                                             dtypes=dtypes, cpu_dtypes=cpu_dtypes, has_inplace=False,
+                                             decorators=decorators, half_tol=half_tol,
+                                             win_float_tol=win_float_tol)
+        self.op_name = op_name
+        self.subtest_name = subtest_name
+        if self.subtest_name and op_name[-1] != '_':
+            self.subtest_name = '_' + self.subtest_name
+        self.ctor = _small_3d if ctor is None else ctor
+        self.arg_ctor = (lambda t, d: []) if arg_ctor is None else arg_ctor
+        self.dtypes = _types if dtypes is None else dtypes
+        self.cpu_dtypes = [] if cpu_dtypes is None else cpu_dtypes
+        self.has_inplace = has_inplace
+        self.decorators = [] if decorators is None else decorators
```

Comment:
nit: The defaults for `ctor`, `dtypes`, `cpu_dtypes`, `decorators` should appear in the `__init__` constructor.

If the linter complains, then that's because any defaults cannot be a list. Although the defaults can't be lists, they can still be (and should be) tuples instead.

I'm fine with handling this in a follow-up if you're trying to avoid merge conflicts

Reply:
You're correct that I moved this due to lint errors. I see there's an easy way to convert a list to a tuple, didn't realize that was an option. I'd probably move that to a followup PR though.